### PR TITLE
[hotfix][docs] Fix minor grammar and spelling mistakes

### PR DESCRIPTION
**Pull request by RyanSkraba**
## What is the purpose of the change

While learning and reviewing the code around Flink sources, I collected some typos and minor English grammar fixes.

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`:  **no**
  - The serializers:  **no**
  - The runtime per-record code paths (performance sensitive):  **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper:  **no**
  - The S3 file system connector:  **no**

## Documentation

  - Does this pull request introduce a new feature?  **no**
  - If yes, how is the feature documented? **not applicable**

### CODE REVIEW

File: flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSourceBuilder.java
Comment (XComp) [inline flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSourceBuilder.java:200]:
```suggestion
     * Specify from which offsets the KafkaSource should start consuming from by providing an {@link
```

File: flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/cursor/StopCursor.java
Comment (XComp) [inline flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/cursor/StopCursor.java:38]:
```suggestion
 * serialized into split, the implementation for this interface should be well considered. It's not
 * recommended to add extra internal state for this implementation.
```

**Review by XComp — CHANGES_REQUESTED** [review]
Thanks @RyanSkraba for this PR. One typo needs to be fixed. See my comments below. Please rebase the branch to prepare it to be merged.

**Comment by RyanSkraba** [comment]
Rebased, squashed into one commit and fixed merge conflicts.  Can you take a look?

**Review by XComp — APPROVED** [review]
Thanks for creating the PR. LGTM :+1:

**Comment by XComp** [comment]
Only documentation was affected. I'm gonna merge right away without waiting for CI.

**Pull request merged**
